### PR TITLE
Update dependency + common module

### PR DESCRIPTION
**Pull request by danybony**
* Update some dependencies, gradle wrapper and build tools
* Create a `common` module, which will contain files shared with the Android Things module
* Move the first 2 files as an example of what I'll do in a next PR (will move basically everything in `common`)

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -1,21 +1,42 @@
 buildscript {
     repositories {
         jcenter()
+        maven {
+            url 'https://maven.google.com/'
```
Comment (takecare) [inline android/build.gradle]: isn't this the same as `google()`? or is this just for compatibility?

File: android/common/build.gradle [hunk 2]
```diff
@@ -0,0 +1,35 @@
+apply plugin: 'com.android.library'
+
+android {
+    compileSdkVersion 27
+    buildToolsVersion "27.0.3"
```
Comment (takecare) [inline android/common/build.gradle]: `''` and for the others too 😄

File: android/common/build.gradle [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+
```
Comment (takecare) [inline android/common/build.gradle]: 👾

File: android/build.gradle (hunk 1 above)
Comment (danybony) [reply android/build.gradle]: yeah, for compatibility https://stackoverflow.com/questions/46467561/difference-between-google-and-maven-url-https-maven-google-com

File: android/common/build.gradle (hunk 2 above)
Comment (danybony) [reply android/common/build.gradle]: Good catch! Will fix

**Pull request merged**